app/models/govuk_publishing_components/component_doc_resolver.rb

```diff
@@ -0,0 +1,41 @@
+module GovukPublishingComponents
+  class ComponentDocResolver
+    def get(id)
+      component = fetch_component_doc(id)
+      build(component)
+    end
+
+    def all
+      fetch_component_docs.map { |component| build(component) }
+    end
+
+  private
+
+    def build(component)
+      fixtures = component[:fixtures].map { |id, data|
+        ComponentFixture.new(id.to_s, data)
+      }
+
+      ComponentDoc.new(component[:id],
+                       component[:name],
+                       component[:description],
+                       component[:body],
+                       component[:accessibility_criteria],
+                       fixtures)
+    end
+
+    def fetch_component_docs
+      doc_files = Rails.root.join("app", "views", "components", "docs", "*.yml")
+      Dir[doc_files].sort.map { |file| parse_documentation(file) }
+    end
+
+    def fetch_component_doc(id)
```

Comment:
I want to make it faster, and memoising makes sense. As this is predominantly a tool for developing components we need to balance that with the problem of too much caching hiding local changes.